src/position.cpp

```diff
@@ -2845,6 +2854,36 @@ bool Position::is_immediate_game_end(Value& result, int ply) const {
       }
   }
 
+  // Collinear-n
+  if (collinear_n() > 0) {
+      Bitboard allPieces = connectPieces;
+      for (Direction d : var->connect_directions) {
+          Bitboard b = allPieces;
+          while (b) {
+              Square s = pop_lsb(b);
+
+              int total_count = 1; // Start with the current piece
+
+              // Check in both directions
+              for (int sign = -1; sign <= 1; sign += 2) {
+                  Bitboard shifted = shift(sign * d, square_bb(s));
+                  while (shifted) {
```

Comment:
I gave some thought to how it would work with connectDirections, especially the diagonal ones on a non-square board, and couldn't come up with something reasonable.